pkg/apis/core/validation/validation.go

```diff
@@ -1533,6 +1533,34 @@ func validateStorageOSPersistentVolumeSource(storageos *core.StorageOSPersistent
 	return allErrs
 }
 
+// ValidatePVSecretReference check whether provided SecretReference object is valid in terms of secret name and namespace.
+// This also take a flag to validate the secret name against DNSSubDomainformat (<253char) and DNS1123Label (<63char) format
+// as its required by some PV sources today, ex: CSI. This flag can help on racheting validation of secretRef field.
+func ValidatePVSecretReference(secretRef *core.SecretReference, allowDNSSubDomainFormat bool, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if len(secretRef.Name) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("name"), ""))
+	} else {
+		if !allowDNSSubDomainFormat {
+			for _, msg := range validation.IsDNS1123Label(secretRef.Name) {
```

Comment:
Why don't we use `ValidateDNS1123Label` here?

Reply:
sure, we can